src/sentry/tasks/post_process.py

```diff
@@ -501,3 +496,49 @@ def plugin_post_process_group(plugin_slug, event, **kwargs):
         _with_transaction=False,
         **kwargs,
     )
+
+
+@instrumented_task(
+    name="sentry.tasks.post_process.index_similarity",
+    queue="similarity.index",
+    time_limit=120,
+    soft_time_limit=110,
+)
+def index_similarity(project_id: int, group_id: int, cache_key: str):
+    """
+    Index the event into similarity. This task is responsible for
+    dropping the event payload from processing store, and is isolated
+    from the rest of post_process due to stability problems of
+    similarity.
+    """
+    from sentry.eventstore.models import Event
+    from sentry.eventstore.processing import event_processing_store
+    from sentry.models import EventDict, Group, Organization, Project
+
+    data = event_processing_store.get(cache_key)
+    set_current_event_project(data["project"])
+
+    # Delete key now such that timing out task does not cause
+    # processing store to leak.
+    with metrics.timer("tasks.post_process.delete_event_cache"):
+        event_processing_store.delete_by_key(cache_key)
```

Comment:
Do you think we would use 120 seconds before getting here? Should we have an except for `SoftTimeLimitExceeded that removes the cache key?

Reply:
try-finally is an interesting proposition, but I would expect to arrive at this line within milliseconds. I purposefully delete the key before doing the bulk of the work in this celery task.

Reply:
I slept over this and decided it'd probably be best to avoid interfering with celery timeouts, as that can cause backlogs which also result in leaking processingstore.